### PR TITLE
Feature/decision_paper

### PR DESCRIPTION
**Pull request by ralph-lange**
Not the final version for the Embedded SIG meeting, but a basis for the OFERA-internal discussion.

### CODE REVIEW

File: _docs/client_library/decision_paper/index.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+* Similar to parameters
+
+Logging (eProsima)
+
+* Optimized implementation for MCU
+
+Time / Clock and Timers (eProsima, Bosch, Acutronic Robotics)
+
+* Bosch will analyze rcl time and clock interface
+* Synchronization with microprocessor -- message types already available in micro-XRCE-DDS
```
Comment (imuguruza) [inline _docs/client_library/decision_paper/index.md]: Being picky is think it should be "Micro XRCE-DDS" as used previously in the document

Comment (ralph-lange) [reply _docs/client_library/decision_paper/index.md]: Fixed it.

File: _docs/client_library/decision_paper/index.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+* Is a two-phase approach -- allow dynamic allocation in some initialization phase but not in a later run phase -- possible? How much effort is it to implement? Would such an approach be acceptable for some safety-certified implementation?
+
+As a first experiment, we implemented a simple node and subscriber directly against the rcl in the C programming language and counted the allocations and frees. The code can be found at [https://github.com/micro-ROS/micro-ROS_experiments/tree/experiment/measure_allocations/rcl_int32_subscriber](https://github.com/micro-ROS/micro-ROS_experiments/tree/experiment/measure_allocations/rcl_int32_subscriber).
+
+In detail, we counted the calls of the standard C memory functions (malloc, realloc, free) and the calls of the functions of the default allocator (which uses the standard C memory functions) in rcutils/allocator.c
+
+With the Micro-XRCE-DDS middleware, we obtained the following numbers:
```
Comment (BorjaOuterelo) [inline _docs/client_library/decision_paper/index.md]:
```suggestion
With the Micro XRCE-DDS middleware, we obtained the following numbers:
```

File: _docs/client_library/decision_paper/index.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+![Allocations with Micro-XRCE-DDS](allocations_Micro-XRCE-DDS.png)
```
Comment (BorjaOuterelo) [inline _docs/client_library/decision_paper/index.md]:
```suggestion
![Allocations with Micro XRCE-DDS](allocations_Micro-XRCE-DDS.png)
```

File: _docs/client_library/decision_paper/index.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+* Similar to parameters
+
+Logging (eProsima)
+
+* Optimized implementation for MCU
+
+Time / Clock and Timers (eProsima, Bosch, Acutronic Robotics)
+
+* Bosch will analyze rcl time and clock interface
+* Synchronization with microprocessor -- message types already available in Micro-XRCE-DDS
```
Comment (BorjaOuterelo) [inline _docs/client_library/decision_paper/index.md]:
```suggestion
* Synchronization with microprocessor -- message types already available in Micro XRCE-DDS
```

File: _docs/client_library/decision_paper/index.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+* To be developed in a second step in the micro-ROS Turtlebot demo.
+
+TF (Bosch)
+
+* To be developed in a third step in the micro-ROS Turtlebot demo.
+
+Diagnostics (Bosch)
+
+* To be developed in the second step in the micro-ROS Turtlebot demo.
+* Liveliness of node: Introduce mechanism in Micro-XRCE-DDS similar to standard DDS?
```
Comment (BorjaOuterelo) [inline _docs/client_library/decision_paper/index.md]:
```suggestion
* Liveliness of node: Introduce mechanism in Micro XRCE-DDS similar to standard DDS?
```

**Pull request merged**